pkg/cloud/metadata/metadata.go

```diff
@@ -136,11 +136,24 @@ func newImmutableProvider(provider MetadataProvider, name string) *immutableProv
 }
 
 func NewMetadata() *Metadata {
-	return &Metadata{
+	defaultMetadata := &Metadata{
 		providers: []MetadataProvider{
 			newImmutableProvider(&ENVMetadata{}, "env"),
 		},
 	}
+	lm, err := NewLingJunMetadata()
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			klog.V(3).InfoS("lingjun metadata file does not exist, use env metadata only", "file", LingjunConfigFile)
+		} else {
+			klog.ErrorS(err, "Failed to read lingjun metadatafile", "file", LingjunConfigFile)
+		}
+		return defaultMetadata
+	}
```

Comment:
OK, but I'd prefer these code to also go into `NewLingJunMetadata`, so that we can have test coverage over it.